apps/sel4test-tests/src/tests/pagetables.c

```diff
@@ -508,3 +511,474 @@ test_pagetable_arm(env_t env)
+
+    return sel4test_get_result();
+}
+DEFINE_TEST(PT0012, "Test that overmapping large pages is not permitted", test_overmap_large, true)
+
+static int test_overmap_section(env_t env)
+{
+    seL4_Word map_addr = 0x10000000;
+    cspacepath_t path;
+    int error;
+
+    seL4_CPtr pd = vka_alloc_object_leaky(&env->vka, seL4_ARM_PageDirectoryObject, 0);
+    seL4_CPtr section = vka_alloc_object_leaky(&env->vka, seL4_ARM_SectionObject, 0);
+    seL4_CPtr section2 = vka_alloc_object_leaky(&env->vka, seL4_ARM_SectionObject, 0);
+    test_assert(pd != 0);
+    test_assert(section != 0);
+    test_assert(section2 != 0);
+
+    seL4_ARM_ASIDPool_Assign(env->asid_pool, pd);
+
+    /* map frame into the page table */
+    error = seL4_ARM_Page_Map(section, pd, map_addr, seL4_AllRights, seL4_ARM_Default_VMAttributes);
+    test_error_eq(error, seL4_NoError);
+
+    /* Remap the frame with readonly permissions */
+    error = seL4_ARM_Page_Map(section, pd, map_addr, seL4_NoWrite, seL4_ARM_Default_VMAttributes);
+    test_error_eq(error, seL4_NoError);
+
+    /* Try to overmap the existing mapping*/
+    error = seL4_ARM_Page_Map(section2, pd, map_addr, seL4_NoWrite, seL4_ARM_Default_VMAttributes);
+    test_error_eq(error, seL4_DeleteFirst);
+
+    return sel4test_get_result();
+}
+DEFINE_TEST(PT0013, "Test that overmapping sections is not permitted", test_overmap_section, true)
+
+static int test_overmap_supersection(env_t env)
+{
+    seL4_Word map_addr = 0x10000000;
+    cspacepath_t path;
+    int error;
+
+    seL4_CPtr pd = vka_alloc_object_leaky(&env->vka, seL4_ARM_PageDirectoryObject, 0);
+    seL4_CPtr supersection = vka_alloc_object_leaky(&env->vka, seL4_ARM_SuperSectionObject, 0);
+    seL4_CPtr supersection2 = vka_alloc_object_leaky(&env->vka, seL4_ARM_SuperSectionObject, 0);
+    test_assert(pd != 0);
+    test_assert(supersection != 0);
+    test_assert(supersection2 != 0);
+
+    seL4_ARM_ASIDPool_Assign(env->asid_pool, pd);
+
+    /* map frame into the page table */
+    error = seL4_ARM_Page_Map(supersection, pd, map_addr, seL4_AllRights, seL4_ARM_Default_VMAttributes);
+    test_error_eq(error, seL4_NoError);
+
+    /* Remap the frame with readonly permissions */
+    error = seL4_ARM_Page_Map(supersection, pd, map_addr, seL4_NoWrite, seL4_ARM_Default_VMAttributes);
+    test_error_eq(error, seL4_NoError);
+
+    /* Try to overmap the existing mapping*/
+    error = seL4_ARM_Page_Map(supersection2, pd, map_addr, seL4_NoWrite, seL4_ARM_Default_VMAttributes);
+    test_error_eq(error, seL4_DeleteFirst);
+
+    return sel4test_get_result();
+}
+DEFINE_TEST(PT0014, "Test that overmapping super sections is not permitted", test_overmap_supersection, true)
+
+#elif defined(CONFIG_ARCH_AARCH64)
+static int test_overmap_small(env_t env)
+{
+    seL4_Word map_addr = 0x10000000;
+    cspacepath_t path;
+    int error;
+
+    seL4_CPtr pgd = vka_alloc_object_leaky(&env->vka, seL4_ARM_PageGlobalDirectoryObject, 0);
```

Comment:
Since Kent's commit to unify the page table objects, this should be `seL4_ARM_VSpaceObject` and the others just `seL4_ARM_PageTable`.

However, backward compatible defines are in `libsel4/sel4_arch_include/aarch64/sel4/sel4_arch/deprecated.h`. (But they're missing from `libsel4/sel4_arch_include/aarch32/sel4/sel4_arch/deprecated.h`, not sure if that's intentional.)